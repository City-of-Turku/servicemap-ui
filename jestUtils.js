import React from 'react';
import { render } from '@testing-library/react';
import configureStore from 'redux-mock-store';
import { ThemeProvider } from '@mui/material/styles';
import { IntlProvider } from 'react-intl';
import { Provider } from 'react-redux';
import themes from './src/themes';
import finnish from './src/i18n/fi';

const mockStore = configureStore([]);

// Mock props for intl provider
const intlMock = {
  locale: 'fi',
  messages: finnish,
  wrapRichTextChunksInFragment: false,
};

const Providers = (mockState) => ({ children }) => {
  const store = mockStore(mockState);
  return (
    <Provider store={store}>
      <IntlProvider {...intlMock}>
        <ThemeProvider theme={themes.SMTheme}>
          {children}
        </ThemeProvider>
      </IntlProvider>
    </Provider>
  );
}

export const getRenderWithProviders = mockState => component => render(component, { wrapper: Providers(mockState) });