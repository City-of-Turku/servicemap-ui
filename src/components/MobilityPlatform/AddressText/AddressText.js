import React from 'react';
import { Typography } from '@material-ui/core';
import PropTypes from 'prop-types';
import useLocaleText from '../../../utils/useLocaleText';

const AddressText = ({ classes, intl, addressObj }) => {
  const getLocaleText = useLocaleText();

  const renderAddressText = (messageId, value, props = {}) => (
    <div {...props}>
      <Typography component="p" variant="body2">
        {intl.formatMessage({ id: messageId }, { value })}
      </Typography>
    </div>
  );

  return (
    <>
      {renderAddressText('mobilityPlatform.content.address', getLocaleText(addressObj), { className: classes.margin })}
    </>
  );
};

AddressText.propTypes = {
  classes: PropTypes.objectOf(PropTypes.any).isRequired,
  intl: PropTypes.objectOf(PropTypes.any).isRequired,
  addressObj: PropTypes.objectOf(PropTypes.any),
};

AddressText.defaultProps = {
  addressObj: {},
};

export default AddressText;
