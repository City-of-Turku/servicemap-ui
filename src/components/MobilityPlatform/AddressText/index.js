import { withStyles } from '@material-ui/core';
import { injectIntl } from 'react-intl';
import AddressText from './AddressText';
import styles from './styles';

export default withStyles(styles)(injectIntl(AddressText));
