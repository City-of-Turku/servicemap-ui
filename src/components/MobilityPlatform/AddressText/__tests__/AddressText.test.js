// Link.react.test.js
import React from 'react';
import AddressText from '../index';
import { initialState } from '../../../../redux/reducers/user';
import { getRenderWithProviders } from '../../../../../jestUtils';

const mockProps = {
  addressObj: {
    address_fi: 'Testikatu',
    address_en: 'Test street',
    address_sv: 'Test gata',
  },
};

const renderWithProviders = getRenderWithProviders({
  user: initialState,
});

describe('<AddressText />', () => {
  it('should match snapshot', () => {
    const { container } = renderWithProviders(<AddressText {...mockProps} />);
    expect(container).toMatchSnapshot();
  });

  it('does show text correctly', () => {
    const { container } = renderWithProviders(<AddressText {...mockProps} />);

    const p = container.querySelectorAll('p');
    expect(p[0].textContent).toContain(`Osoite: ${mockProps.addressObj.address_fi}`);
  });
});
