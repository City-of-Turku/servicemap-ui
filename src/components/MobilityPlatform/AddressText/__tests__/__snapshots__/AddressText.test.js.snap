// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddressText /> should match snapshot 1`] = `
<div>
  <div
    class="injectIntl(AddressText)-margin-1"
  >
    <p
      class="MuiTypography-root MuiTypography-body2"
    >
      Osoite: Testikatu
    </p>
  </div>
</div>
`;
